database/src/main/java/com/firebase/ui/database/FirebaseIndexRecyclerAdapter.java

```diff
@@ -59,8 +59,11 @@
  * </pre>
  *
  * @param <T>  The Java class that maps to the type of objects stored in the Firebase location.
- * @param <VH> The ViewHolder class that contains the Views in the layout that is shown for each object.
+ * @param <VH> The ViewHolder class that contains the Views in the layout that is shown for each
+ *             object.
+ * @deprecated use {@link com.firebase.ui.database.adapter.FirebaseIndexRecyclerAdapter} instead
  */
+@Deprecated
```

Comment:
@samtstern Yeah, the permuted arguments was a refactoring mistake.

@samtstern I would never needlessly deprecate something so here the missing key point in that list above: we are now implementing `ChangeEventListener` directly. If you diff the two files, you'll notice that the protected methods that forward calls from `ChangeEventListener` are now public, thus breaking backwards compatibility. I wrote a blurb in my original post about why we should directly implement the interface, but I'll reiterate the main point here: since people can access `FirebaseArray` directly now, it makes no sense to have an anonymous listener forwarding calls to identical methods whose only difference to the interface is being protected. If you look at what I did in the old adapters, we have to store the listener to be able to remove it later and it would be such a pain for devs if they wanted change listeners midway or something. Also, it's just confusing in general if devs try to implement `ChangeEventListener` and they get compile errors because the forwarding methods are protected.